src/SLCore/ISLCoreErrorLoggerFactory.cs

```diff
@@ -0,0 +1,101 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel.Composition;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.Threading;
+using SonarLint.VisualStudio.Core;
+
+namespace SonarLint.VisualStudio.SLCore;
+
+internal interface ISLCoreErrorLoggerFactory
+{
+    ISLCoreErrorLogger Create(StreamReader errorStream);
+}
+
+internal interface ISLCoreErrorLogger : IDisposable
+{
+}
+
+[Export(typeof(ISLCoreErrorLoggerFactory))]
+[PartCreationPolicy(CreationPolicy.Shared)]
+internal class ISlCoreErrorLoggerFactory : ISLCoreErrorLoggerFactory
+{
+    private readonly ILogger logger;
+    private readonly IThreadHandling threadHandling;
+
+    public ISlCoreErrorLoggerFactory(ILogger logger, IThreadHandling threadHandling)
+    {
+        this.logger = logger;
+        this.threadHandling = threadHandling;
+    }
+
+    public ISLCoreErrorLogger Create(StreamReader errorStream)
+    {
+        return new ErrorLogger(logger, threadHandling, errorStream);
+    }
+
+    private class ErrorLogger : ISLCoreErrorLogger
+    {
+        private readonly ILogger logger;
+        private readonly IThreadHandling threadHandling;
+        private readonly CancellationTokenSource cancellationTokenSource;
+
+        public ErrorLogger(ILogger logger, IThreadHandling threadHandling, StreamReader streamReader)
+        {
+            this.logger = logger;
+            this.threadHandling = threadHandling;
+            cancellationTokenSource = new CancellationTokenSource();
+            ReadErrorStreamInBackground(streamReader);
+        }
+
+        private void ReadErrorStreamInBackground(StreamReader errorStream)
+        {
+            threadHandling.RunOnBackgroundThread(async () =>
+            {
+                await ReadAsync(errorStream);
+                return 0;
+            }).Forget();
+        }
+
+        private async Task ReadAsync(TextReader errorStream)
+        {
+            while (!cancellationTokenSource.IsCancellationRequested)
+            {
+                var line = await errorStream.ReadLineAsync();
+                if (string.IsNullOrEmpty(line))
+                {
+                    continue;
+                }
+
+                logger.LogVerbose("[SLCORE-ERR]" + line);
```

Comment:
`continue` should be avoided where possible, it makes the logic hard to follow
```suggestion
                if (!string.IsNullOrEmpty(line))
                {
                    logger.LogVerbose("[SLCORE-ERR]" + line);
                }
```

Reply:
I disagree. It is the case when you have many statements inside of the If body and there's a lot of branching, but here it actually makes the intent of skipping the iteration explicit. if I saw the suggested variant, I would think that someone forgot to finish the method. The good alternative would be to move the IsNullOrEmpty condition to the while statement, but it won't look great in this particular case due to variable assignment.

Reply:
I agree that in this case having "continue" makes it more obvious of the intentionality of the skipping.